packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Primitives/CSharpType.cs

```diff
@@ -257,7 +257,7 @@ private CSharpType GetPropertyInitializationType()
             {
                 if (IsReadOnlyMemory)
                 {
-                    return new CSharpType(typeof(ReadOnlyMemory<>), Arguments);
+                    return new CSharpType(ElementType.FrameworkType.MakeArrayType());
```

Comment:
We can revert the change here and use this logic in MrwSerializationTypeDefinition. ReadOnlyMemory makes sense as the property initialization type in general, but in the serialization case we need an array.